File: 38-count-and-say/38-count-and-say.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
        if(n==1)
            return "1";
        let prev=countAndSay(n-1);
        let i=0;
        let res='';
        while(i<prev.length){
            let j=i+1;
            while(j<prev.length&&prev[j]==prev[i])
                j++;
            res+=(j-i)+prev[i];
            i=j;
        }
    return res;
};

File: 38-count-and-say/38-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        if(n==1)
            return "1";
        string prev=countAndSay(n-1);
        int i=0;
        string res;
        while(i<prev.size()){
            int j=i+1;
            while(j<prev.size()&&prev[j]==prev[i])
                j++;
            int diff=j-i;
            res.push_back(diff+'0');
            res.push_back(prev[i]);
            i=j;
        }
        return res;
    }
};

    // 1
    // 11
    // 21
    // 1211
    // 111221